codegen/projections/high_score_service/sig/high_score_service/endpoint.rbs

```diff
@@ -14,7 +14,7 @@ module HighScoreService
     end
 
     class Provider
-      def resolve_endpoint: (Params params) -> Hearth::RulesEngine::Endpoint
+      def resolve_endpoint: (Params params) -> Hearth::Endpoints::Endpoint
```

Comment:
After seeing this, `Hearth::Endpoints::Endpoint` seems weird. I think `Hearth::EndpointRules::Endpoint` to make clear that it's an endpoint object as defined in the endpoint ruels, but I'll defer to you.